### PR TITLE
UCP/EP/CLOSE: Make close EP discard lanes directly

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

File: src/ucp/core/ucp_ep.c [hunk 1]
```diff
@@ -948,41 +948,112 @@ ucs_status_ptr_t ucp_ep_close_nb(ucp_ep_h ep, unsigned mode)
...
+        .ep_destroy          = (uct_ep_destroy_func_t)ucs_empty_function,
+        .ep_get_address      = (uct_ep_get_address_func_t)ucs_empty_function_return_ep_timeout
+    }
+};
+
+static uct_ep_t ucp_failed_tl_ep = {
+    .iface = &ucp_failed_tl_iface
+};
+
+void ucp_ep_flush_cancel(ucp_ep_h ep, ucs_status_t status)
```
Comment (yosefe) [inline src/ucp/core/ucp_ep.c:989]: ucp_ep_discard_lanes

File: src/ucp/core/ucp_ep.c [hunk 2]
```diff
@@ -948,41 +948,112 @@ ucs_status_ptr_t ucp_ep_close_nb(ucp_ep_h ep, unsigned mode)
...
+{
+    ucp_lane_index_t lane;
+
+    for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
```
Comment (yosefe) [inline src/ucp/core/ucp_ep.c:993]: maybe add assert that FAILED flag is not set

Comment (Artemy-Mellanox) [reply src/ucp/core/ucp_ep.c:993]: but when called from ucp_worker_iface_err_handle_progress() FAILED is set

Comment (yosefe) [reply src/ucp/core/ucp_ep.c:993]: ok

**Review by yosefe — APPROVED** [review]

**Review by evgeny-leksikov — APPROVED** [review]

**Pull request merged**